internal/containerd/restart_unix.go

```diff
@@ -22,35 +22,139 @@ package containerd
 import (
 	"fmt"
 	"log/slog"
+	"os"
+	"os/exec"
+	"regexp"
 	"syscall"
 
 	"github.com/mitchellh/go-ps"
 )
 
 var psProcesses = ps.Processes
 
-type restarter struct{}
+type defaultRestarter struct{}
 
-func NewRestarter() Restarter {
-	return restarter{}
+func NewDefaultRestarter() Restarter {
```

Comment:
I don't know about you, but I kind of expected the `Restarter` interface to be in this file.
At least I think it is somewhat misplaced in `configure.go`.

wdyt?

Reply:
I'm speculating that the `Restarter` interface was placed there in anticipation of use between the `restart_unix.go` and `restart_windows.go` files.  I don't have a strong opinion on where it should live, but could see it worth revisiting.  Follow-up material for when/if it bugs us again :)